test/application_system_test_case.rb

```diff
@@ -21,9 +21,13 @@ class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
   AXE_WITHIN_SELECTOR = "body"
 
   def with_preview(preview_name)
-    component_uri = self.class.name.gsub("Test", "").gsub("Integration", "").underscore
+    component_name = self.class.name.gsub("Test", "").gsub("Integration", "")
+    status = ""
+    status = "beta/" if component_name.include?("Beta")
+    status = "alpha/" if component_name.include?("Alpha")
+    component_uri = component_name.gsub("Alpha", "").gsub("Beta", "").underscore
 
-    visit("/rails/view_components/primer/#{component_uri}/#{preview_name}")
+    visit("/rails/view_components/primer/#{status}#{component_uri}/#{preview_name}")
```

Comment:
This seems like a fragile way of deriving the preview path 😬 out of the scope of this PR to refactor but I'll make a note of it

Reply:
Yeah, I thought the same - let's do it in this PR and then it's done! Initially I was just trying to get my stuff working.

Maybe I could change this to [starts_with](https://apidock.com/ruby/String/start_with%3F) and then at least we wouldn't be hit with a problem if someone creates a `BetaAlphabetSelector` although it would still fail if someone created `AlphabetSelector`.

If we want to go further, I could use some regex to match `Alpha` and `Beta` and a follow-up capital letter?

Reply:
I was thinking more that getting the data from the class is a bit fragile, but is that pretty normal in Rails? Either way no harm in refactoring this with a bit of regex. Would something like this work:

```rb
match = /^(Alpha|Beta)([A-Z])/.match(component_name)
status = match[1]
statusPath = status ? "#{status.downcase}/" : ""
component_uri = component_name.gsub(status, "").underscore
```

Reply:
> I was thinking more that getting the data from the class is a bit fragile, but is that pretty normal in Rails?

No, I wouldn't say normal. There's another option (some sort of look-up based on https://github.com/primer/view_components/blob/2ce820192a5fa027a9bc1bb806bf63475758369d/static/statuses.json#L1), but it would break when things do not match our expectations around naming conventions (which some components seem not to). I think the regex solution is better than my original solution and anything wider would need some conversations to work out the correct approach (and some remediation work to move around some of the existing preview files etc).

I'm happy to resolve this if you're happy with the code? I made a few changes from your suggestion above.